test/spanner.ts

```diff
@@ -1390,6 +1390,89 @@ describe('Spanner with mock server', () => {
       }
     });
 
+    it('should return the results correctly when last field is present in PartialResultSet for query', async () => {
+      // Setup a query result with more than maxQueued (10) PartialResultSets.
+      // None of the PartialResultSets include a resume token.
+      const sql = 'SELECT C1 FROM TestTable';
+      const fields = [
+        protobuf.StructType.Field.create({
+          name: 'C1',
+          type: protobuf.Type.create({code: protobuf.TypeCode.STRING}),
+        }),
+      ];
+      const metadata = new protobuf.ResultSetMetadata({
+        rowType: new protobuf.StructType({
+          fields,
+        }),
+      });
+      const results: PartialResultSet[] = [];
+      for (let i = 0; i < 2; i++) {
+        results.push(
+          PartialResultSet.create({
+            metadata,
+            values: [{stringValue: `V${i}`}],
+            last: i === 1,
+          }),
+        );
+      }
```

Comment:
result simulation should be in mockspanner.ts file

ref: https://github.com/googleapis/nodejs-spanner/blob/main/test/mockserver/mockspanner.ts#L1283